forge-game/src/main/java/forge/game/ability/effects/TextBoxExchangeEffect.java

```diff
@@ -0,0 +1,124 @@
+package forge.game.ability.effects;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Table.Cell;
+
+import forge.game.Game;
+import forge.game.ability.SpellAbilityEffect;
+import forge.game.card.Card;
+import forge.game.card.CardState;
+import forge.game.card.CardCopyService;
+import forge.game.event.GameEventCardStatsChanged;
+import forge.game.keyword.KeywordInterface;
+import forge.game.keyword.KeywordsChange;
+import forge.game.replacement.ReplacementEffect;
+import forge.game.spellability.SpellAbility;
+import forge.game.staticability.StaticAbility;
+import forge.game.trigger.Trigger;
+
+import java.util.List;
+
+/**
+ * Exchanges text boxes between two creatures.
+ */
+public class TextBoxExchangeEffect extends SpellAbilityEffect {
+    @Override
+    protected String getStackDescription(final SpellAbility sa) {
+        final List<Card> tgtCards = getTargetCards(sa);
+        Card c1;
+        Card c2;
+        if (tgtCards.size() == 1) {
+            c1 = sa.getHostCard();
+            c2 = tgtCards.get(0);
+        } else {
+            c1 = tgtCards.get(0);
+            c2 = tgtCards.get(1);
+        }
+        return c1 + " exchanges text box with " + c2 + ".";
+    }
+
+    @Override
+    public void resolve(final SpellAbility sa) {
+        final List<Card> tgtCards = getTargetCards(sa);
+        final Card host = sa.getHostCard();
+        if (tgtCards.isEmpty()) {
+            return;
+        }
+
+        final Card c1;
+        final Card c2;
+        if (tgtCards.size() == 1) {
+            c1 = host;
+            c2 = tgtCards.get(0);
+        } else {
+            c1 = tgtCards.get(0);
+            c2 = tgtCards.get(1);
+        }
+
+        // snapshot the original text boxes before modifying
+        final Card copy1 = CardCopyService.getLKICopy(c1);
+        final Card copy2 = CardCopyService.getLKICopy(c2);
+
+        final Game game = host.getGame();
+        final long ts = game.getNextTimestamp();
+
+        swapTextBox(c1, copy2, ts);
+        swapTextBox(c2, copy1, ts);
+
+        game.fireEvent(new GameEventCardStatsChanged(c1));
+        game.fireEvent(new GameEventCardStatsChanged(c2));
+    }
+
+    private static void swapTextBox(final Card to, final Card from, final long ts) {
+        to.getChangedCardTraitsByText().clear();
```

Comment:
imo clearing the CardTraitsByText might be dangerous

and I wish there was a way to code this via Static Layers @tool4ever